mailjet_client_test.go

```diff
@@ -178,3 +178,43 @@ func TestSendMail(t *testing.T) {
 	}
 	fmt.Printf("Data: %+v\n", res)
 }
+
+func TestSendMailV31(t *testing.T) {
+	m := NewMailjetClient(
+		os.Getenv("MJ_APIKEY_PUBLIC"),
+		os.Getenv("MJ_APIKEY_PRIVATE"))
+
+	var data []resources.Sender
+	count, _, err := m.List("sender", &data)
+	if err != nil {
+		t.Fatal("Unexpected error:", err)
+	}
+	if count < 1 || data == nil {
+		t.Fatal("At least one sender expected in the test account!")
+	}
+
+	param := []InfoMessagesV31{InfoMessagesV31{
+		From: &RecipientV31{
+			Email: data[0].Email,
+			Name:  data[0].Name,
+		},
+		To: &RecipientsV31{
+			RecipientV31{
+				Email: data[0].Email,
+			},
+		},
+		Subject:  "Send API testing",
+		TextPart: "SendMail is working !",
+	},
+	}
+
+	messages := MessagesV31{Info: param}
+
+	res, err := m.SendMailV31(&messages)
+	if err != nil {
+		t.Fatal("Unexpected error:", err)
+	}
+	if res != nil {
```

Comment:
The HTTP response along with the status code is wrapped inside the wrapper

When the call is successful, all the functions of the driver (including this one), returns a response struct

In case of error, an error is returned
This error contains informations feedback about what happened and the status code